### PR TITLE
Customizable Discord Message and Prompts

### LINKED ISSUE #13: Support other club servers
This bot currently is built for a special channel in our club ([ACM at CSUF](https://acmcsuf.com)) Discord server known as the "Opportunities" channel. There is a desire to share this technology with partner clubs.

### Proposal

We could abstract a way to define sets of opportunities depending on what is relevant to the club.

### PR DESCRIPTION
**Pull request by boushrabettir**
### TODO
1. Create functionality for optional customizable options or both webhook message and PaLM API prompts
2. Ensure functionality works + no bugs occur during multiple testing phases
<br/>
So far the base format has been placed in the TXT files called `message.txt` and `prompt.txt`. Functionality will be placed soon.

### CODE REVIEW

**Comment by boushrabettir** [comment]
`Side Note`: We should not solve these merge conflicts until Fix #18 has been merged into the main branch.

**Comment by boushrabettir** [comment]
Users are not obligated to possess customizations for their messages and prompts. They have the freedom to selectively determine their preferences. In instances where a customized message or prompt is not furnished, the default prompt/message will be established.

**View the following results below:**

image
image

**Comment by boushrabettir** [comment]
**Mentioning this in order to remember before closing this PR**
When this pull request is successfully closed, we can proceed to close Issue #13.

**Comment by boushrabettir** [comment]
Updated the Github web scrapping since the repository was updated (specifically the `README.md`). Customization options seem to be working well.
image

File: .github/workflows/automate.yaml
Comment (EthanThatOneKid) [inline .github/workflows/automate.yaml]: This command is the same command that you run locally to spin up the server.
```suggestion
        run: python main.py --days-needed 2
```

File: db.py
Comment (EthanThatOneKid) [inline db.py:23]: Where is this function used? 
```suggestion
def delete_all_opportunity_type(opp_type: str) -> None:
```

File: .github/workflows/automate.yaml
Comment (EthanThatOneKid) [inline .github/workflows/automate.yaml:1]: Which one of the following is true?
- Your program uses `PROMPTS_PATH`.
- Your program uses `MESSAGE_PATH`.
- Your program uses both of these environment variables. 

What is the purpose of each environment variable used in your program?

File: main.py
Comment (EthanThatOneKid) [inline main.py]: Why is this commented out? What is `asyncio.run`?

**Review by EthanThatOneKid — COMMENTED** [review]
There’s a lot to this PR!

File: main.py
Comment (boushrabettir) [reply main.py]: This is so the workflow doesn't get triggered. `asyncio.run` has been there since the beginning of this project.

File: db.py
Comment (boushrabettir) [reply db.py:23]: This is just a function for debugging, I found it useful to keep it there just in case for anything.

File: .github/workflows/automate.yaml
Comment (boushrabettir) [reply .github/workflows/automate.yaml]: Awesome, thank you. I definitely forgot to update this.

File: .github/workflows/automate.yaml
Comment (boushrabettir) [reply .github/workflows/automate.yaml:1]: The `PROMPTS_PATH` is for the customizable prompt. `MESSAGE_PATH` is for the customizable message. It was mentioned in the meeting, but I could have accidently glossed over this.

File: db.py
Comment (EthanThatOneKid) [reply db.py:23]: > This is just a function for debugging, I found it useful to keep it there just in case for anything.

Feel free to mention this in the documentation comment of that function signature.

File: main.py
Comment (EthanThatOneKid) [reply main.py]: > This is so the workflow doesn't get triggered. `asyncio.run` has been there since the beginning of this project.

To temporarily disable the workflow, I’d recommend commenting out the workflow step itself.

Comment (boushrabettir) [reply main.py]: For sure, will keep that in mind.

File: .github/workflows/automate.yaml
Comment (EthanThatOneKid) [reply .github/workflows/automate.yaml:1]: All good; thanks for clarifying 😎

**Comment by EthanThatOneKid** [comment]
@boushrabettir, is there anything directly blocking this pull request from being merged on your end?

**Comment by boushrabettir** [comment]
> @boushrabettir, is there anything directly blocking this pull request from being merged on your end?

Hi @EthanThatOneKid ! Yes, I have to get your approval directly.

File: .github/workflows/automate.yaml
Comment (EthanThatOneKid) [inline .github/workflows/automate.yaml]: Nit: Resolve [check failure](https://github.com/acmcsufoss/acmOpportunities/actions/runs/5969957671/job/16196735903?pr=25#step:4:21).
```suggestion
#LINT.IF .env.example
```

**Review by EthanThatOneKid — CHANGES_REQUESTED** [review]

File: .github/workflows/automate.yaml
Comment (boushrabettir) [reply .github/workflows/automate.yaml]: Fix completed!

**Review by EthanThatOneKid — APPROVED** [review]
Looks like a lot of thought went into this PR; very nice work!

**Pull request merged**
